### PR TITLE
Refactored builder patter in java API

### PR DESCRIPTION
**Pull request by marcin-cebo**
Added methods to Channel entity for java and kotlin API

### CODE REVIEW

**Review by wkal-pubnub — CHANGES_REQUESTED** [review]

File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/PubNub.kt [hunk 1]
```diff
@@ -319,6 +320,16 @@ interface PubNub :
...
+
+    @Deprecated(
+        replaceWith =
+            ReplaceWith(
+                "publish(message, channel)"
+            ),
+        level = DeprecationLevel.WARNING,
+        message = "Use publish(Object, String) instead",
+    )
     fun publish(): Publish
```
[the remark this answers is not in the document]
Comment (marcin-cebo) [reply pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/PubNub.kt:333]: I will add it.

File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/PubNub.kt [hunk 2]
```diff
@@ -328,7 +339,17 @@ interface PubNub :
...
+
+    @Deprecated(
+        replaceWith =
+            ReplaceWith(
+                "signal(message, channel)"
+            ),
+        level = DeprecationLevel.WARNING,
+        message = "Use signal(Object, String) instead",
+    )
+    fun signal(): com.pubnub.api.endpoints.pubsub.Signal
```
[the remark this answers is not in the document]
Comment (marcin-cebo) [reply pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/PubNub.kt:352]: I will add it.

File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/PubNub.kt [hunk 3]
```diff
@@ -494,6 +515,16 @@ interface PubNub :
...
+
+    @Deprecated(
+        replaceWith =
+            ReplaceWith(
+                "publish(message, channel)"
+            ),
+        level = DeprecationLevel.WARNING,
+        message = "Use publish(Object, String) instead",
+    )
     fun fire(): Publish
```
[the remark this answers is not in the document]
Comment (marcin-cebo) [reply pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/PubNub.kt:528]: I will add it.

File: pubnub-gson/pubnub-gson-impl/src/main/java/com/pubnub/internal/endpoints/pubsub/PublishImpl.java [hunk 4]
```diff
@@ -27,6 +28,13 @@ public PublishImpl(PubNubCore pubnub) {
         this.replicate = true;
     }
 
+    public PublishImpl(PubNubCore pubnub, Object message, String channel) {
+        super(pubnub);
+        this.replicate = true;
```
[the remark this answers is not in the document]
Comment (marcin-cebo) [reply pubnub-gson/pubnub-gson-impl/src/main/java/com/pubnub/internal/endpoints/pubsub/PublishImpl.java:33]: Ok, I will go with:
return new PublishImpl(getPubNubCore()).message(message).channel(channel);

File: pubnub-gson/pubnub-gson-impl/src/main/java/com/pubnub/internal/endpoints/pubsub/SignalImpl.java [hunk 5]
```diff
@@ -21,6 +22,12 @@ public SignalImpl(PubNubCore pubnub) {
         super(pubnub);
     }
 
+    public SignalImpl(PubNubCore pubnub, Object message, String channel) {
+        super(pubnub);
```
[the remark this answers is not in the document]
Comment (marcin-cebo) [reply pubnub-gson/pubnub-gson-impl/src/main/java/com/pubnub/internal/endpoints/pubsub/SignalImpl.java:26]: Ok, I will go with:
return new SignalImpl(getPubNubCore()).message(message).channel(channel);

File: pubnub-kotlin/pubnub-kotlin-impl/src/main/kotlin/com/pubnub/internal/v2/entities/ChannelImpl.kt [hunk 6]
```diff
@@ -1,15 +1,89 @@
...
+    Channel {
+    override fun publish(
+        message: Any,
+        meta: Any?,
+        shouldStore: Boolean?,
+        usePost: Boolean,
+        replicate: Boolean,
+        ttl: Int?
+    ): Publish {
+        return com.pubnub.internal.endpoints.pubsub.PublishImpl(
```
[the remark this answers is not in the document]
Comment (marcin-cebo) [reply pubnub-kotlin/pubnub-kotlin-impl/src/main/kotlin/com/pubnub/internal/v2/entities/ChannelImpl.kt:29]: Ok, I can change it.

**Review by wkal-pubnub — APPROVED** [review]

**Pull request merged**